api/cpp/esp-idf/slint/src/esp_slint.cpp

```diff
@@ -0,0 +1,159 @@
+    m_window = window.get();
+    return window;
+}
+
+std::chrono::milliseconds EspPlatform::duration_since_start() const
+{
+    auto ticks = xTaskGetTickCount();
+    return std::chrono::milliseconds(pdTICKS_TO_MS(ticks));
+}
+
+void EspPlatform::run_event_loop()
+{
+
+    esp_lcd_panel_disp_on_off(panel_handle, true);
+
+    int last_touch_x = 0;
+    int last_touch_y = 0;
+    bool touch_down = false;
+
+    while (true) {
+        slint::platform::update_timers_and_animations();
+
+        bool has_animations = false;
+
+        if (m_window) {
+
+            if (touch_handle) {
+                uint16_t touchpad_x[1] = { 0 };
+                uint16_t touchpad_y[1] = { 0 };
+                uint8_t touchpad_cnt = 0;
+
+                /* Read touch controller data */
+                esp_lcd_touch_read_data(*touch_handle);
+
+                /* Get coordinates */
+                bool touchpad_pressed = esp_lcd_touch_get_coordinates(
+                        *touch_handle, touchpad_x, touchpad_y, NULL, &touchpad_cnt, 1);
+
+                if (touchpad_pressed && touchpad_cnt > 0) {
+                    // ESP_LOGI(TAG, "x: %i, y: %i", touchpad_x[0], touchpad_y[0]);
+                    last_touch_x = touchpad_x[0];
+                    last_touch_y = touchpad_y[0];
+                    m_window->window().dispatch_pointer_move_event(
+                            slint::LogicalPosition({ float(last_touch_x), float(last_touch_y) }));
+                    if (!touch_down) {
+                        m_window->window().dispatch_pointer_press_event(
+                                slint::LogicalPosition(
+                                        { float(last_touch_x), float(last_touch_y) }),
+                                slint::PointerEventButton::Left);
+                    }
+                    touch_down = true;
+                } else if (touch_down) {
+                    m_window->window().dispatch_pointer_release_event(
+                            slint::LogicalPosition({ float(last_touch_x), float(last_touch_y) }),
+                            slint::PointerEventButton::Left);
+                    m_window->window().dispatch_pointer_exit_event();
+                    touch_down = false;
+                }
+            }
+
+            if (std::exchange(m_window->needs_redraw, false)) {
+                auto region = m_window->m_renderer.render(buffer1, size.width);
+                auto o = region.bounding_box_origin();
+                auto s = region.bounding_box_size();
+                if (s.width > 0 && s.height > 0) {
+                    for (int y = o.y; y < o.y + s.height; y++) {
+                        for (int x = o.x; x < o.x + s.width; x++) {
+                            // Swap endianess to big endian
+                            auto px = reinterpret_cast<uint16_t *>(&buffer1[y * size.width + x]);
+                            *px = (*px << 8) | (*px >> 8);
```

Comment:
We need a way to configure this, because that depends on the screen.
Even if we use a Rgb565BE type or something like that.
I guess it should be another parameter of the constructor. (of type [lcd_color_rgb_endian_t](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/peripherals/lcd.html#_CPPv422lcd_color_rgb_endian_t))
Or it could be a build option with Kconfig but imho that's more complicated

Reply:
I'd go for a Kconfig option with the supported pixel types - either as "combobox" or as multiple separate options. We may not want to compile in the "other" code.

But I've said that already before with my suggestion for feature flags.